.github/workflows/test.yml

```diff
@@ -37,12 +41,14 @@ jobs:
         key: venv-${{ matrix.python-version }}-${{ hashFiles('poetry.lock') }}
     - name: Install dependencies
       run: |
+        poetry config experimental.new-installer false
         poetry config virtualenvs.in-project true
         poetry install
 
         poetry run pip install -U pip
 
     - name: Run tests
+      if: matrix.python-version != '3.10.0-beta.2'
```

Comment:
Maybe you can just create a separate step like 

```yml
name: Run lint 
if: matrix.python-version != '3.10.0-beta.2'
run: poetry run flake8 .
```
?

Reply:
It seems more interesting!

Reply:
Done!